gdk/src/atom.rs

```diff
@@ -79,13 +79,13 @@ impl<'a> ToGlibPtrMut<'a, *mut ffi::GdkAtom> for Atom {
     }
 }
 
-impl<'a> ToGlibContainerFromSlice<'a, *mut ffi::GdkAtom> for &'a Atom {
+impl<'a> ToGlibContainerFromSlice<'a, *mut ffi::GdkAtom> for Atom {
     type Storage = (
-        Vec<Stash<'a, ffi::GdkAtom, &'a Atom>>,
+        Vec<Stash<'a, ffi::GdkAtom, Atom>>,
         Option<Vec<ffi::GdkAtom>>,
     );
 
-    fn to_glib_none_from_slice(t: &'a [&'a Atom]) -> (*mut ffi::GdkAtom, Self::Storage) {
+    fn to_glib_none_from_slice(t: &'a [Atom]) -> (*mut ffi::GdkAtom, Self::Storage) {
         skip_assert_initialized!();
 
         let v: Vec<_> = t.iter().map(|s| s.to_glib_none()).collect();
```

Comment:
This seems silly. Can't you just work with `t.as_ptr()` here without copying everything into a new `Vec` first?

Reply:
Would need a `#[repr(transparent)]` on the struct though

Reply:
I didn't look into the implementation in details, i will give it a try later today

Reply:
We can't do that yet because we always generate a `NULL`-terminated array here. We need different traits, which is on my todo list for after the release already.